### PR TITLE
added town & nation contexts

### PR DESCRIPTION
**Pull request by Galster-dev**
#### Description: 

Permission per town/nation is a must-have feature for people doing automation without new plugins. I don't understand why it is not a thing yet. Here's a PR to fix it
____
#### New Nodes/Commands/ConfigOptions: 

none.

____
#### Relevant Towny Issue ticket:

Haven't seen one (surprisingly)

____
- [x] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

File: src/com/palmergames/bukkit/towny/hooks/LuckPermsContexts.java [hunk 1]
```diff
@@ -28,10 +31,14 @@ public class LuckPermsContexts implements ContextCalculator<Player> {
 	private static final String INSIDEOWNPLOT_CONTEXT = "towny:insideownplot";
 	private static final String TOWN_RANK_CONTEXT = "towny:townrank";
 	private static final String NATION_RANK_CONTEXT = "towny:nationrank";
+	private static final String TOWN_CONTEXT = "towny:town";
+	private static final String NATION_CONTEXT = "towny:nation";
 	
 	private static final List<String> booleanContexts = Arrays.asList(RESIDENT_CONTEXT, MAYOR_CONTEXT, KING_CONTEXT, INSIDETOWN_CONTEXT, INSIDEOWNTOWN_CONTEXT, INSIDEOWNPLOT_CONTEXT);
 	private static final String nationRankContext = NATION_RANK_CONTEXT;
 	private static final String TownRankContext = TOWN_RANK_CONTEXT;
```
Comment (Galster-dev) [inline src/com/palmergames/bukkit/towny/hooks/LuckPermsContexts.java:39]: I also noticed incorrect naming convention here. Should I fix this in same PR (anyway doing contexts)

Comment (LlmDl) [reply src/com/palmergames/bukkit/towny/hooks/LuckPermsContexts.java:39]: Sure

Comment (Warriorrrr) [reply src/com/palmergames/bukkit/towny/hooks/LuckPermsContexts.java:39]: These fields have no use, feel free to remove them

Comment (Galster-dev) [reply src/com/palmergames/bukkit/towny/hooks/LuckPermsContexts.java]: Forgot they are private, thought of compatibility issues lol

**Comment by LlmDl** [comment]
I think the only downside to this is that LP will check every context on each hasPermission test (or so I am told.) Adding further contexts will add additional load for servers that enable towny's LP contexts in the config.

Not that it replaces your PR but there is already a node given towny.town.TOWNNAME to all Town members. This could always be given child nodes. 

I will defer to @Warriorrrr on this PR as he is the original author of our LP Contexts and knows more about it than I do.

**Review by Warriorrrr — APPROVED** [review]
Adding these new contexts is probably fine, I'll make a PR that allows server owners to configure what contexts to enable once this is merged.

**Pull request merged**